modules/compactor/compactor.go

```diff
@@ -236,12 +237,16 @@ func (c *Compactor) Combine(dataEncoding string, tenantID string, objs ...[]byte
 		return objs[0], wasCombined, nil
 	}
 
-	spansDiscarded := countSpans(dataEncoding, objs[1:]...)
-	overrides.RecordDiscardedSpans(spansDiscarded, reasonCompactorDiscardedSpans, tenantID)
+	totalDiscarded, discarded := countSpans(dataEncoding, objs[1:]...)
```

Comment:
in the combine function all of the input traces will have the same id. we don't need to change the return of "countSpans" we only need to log the dropped trace id if `totalDiscarded` is greater than 0.

Reply:
I reverted that countSpans only returns `totalDiscarded` and also moved logging code inside of the function.